src/vic3/effect_validation.rs

```diff
@@ -243,6 +244,57 @@ pub fn validate_country_value(
     vd.field_script_value("value", sc);
 }
 
+pub fn validate_create_building(
+    _key: &Token,
+    block: &Block,
+    _data: &Everything,
+    sc: &mut ScopeContext,
+    mut vd: Validator,
+    _tooltipped: Tooltipped,
+) {
+    vd.req_field("building");
+    vd.field_item("building", Item::BuildingType);
+    let building = block.get_field_value("building");
+    vd.field_validated_list("activate_production_methods", |token, data| {
+        data.verify_exists(Item::ProductionMethod, token);
+        if let Some(building) = building {
+            if let Some((_, block, building_item)) =
+                data.get_item::<BuildingType>(Item::BuildingType, building.as_str())
+            {
+                building_item.validate_production_method(token, building, block, data);
+            }
+        }
+    });
+    vd.field_bool("subsidized");
+    vd.field_numeric_range("reserves", 0.0..=1.0);
```

Comment:
There are no `reserves=0.*` in vanilla, but there is one with `=2`. Maybe this is in absolute unit?

Reply:
I got this from the effect documentation. I don't know why there's a reserves=2 or what it's supposed to mean, i figured it was a bug in vanilla.